bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java

```diff
@@ -322,17 +322,24 @@ private static ColumnPrinter makePrinterForConfiguration(ConfigurationFactory co
         Map<String, String> replaced = new HashMap<>();
         properties.forEach((propertyKey, propertyValue) -> {
             Matcher matcher = ENV_PATTERN.matcher(propertyValue);
-            if (!matcher.matches()) {
+            List<String> envNames = new ArrayList<>();
+            while (matcher.find()) {
+                envNames.add(matcher.group(1));
+            }
+            if (envNames.isEmpty()) {
                 replaced.put(propertyKey, propertyValue);
                 return;
             }
-            String envName = matcher.group(1);
-            String envValue = environment.get(envName);
-            if (envValue == null) {
-                onError.accept(propertyKey, format("Configuration property '%s' references unset environment variable '%s'", propertyKey, envName));
-                return;
+            String replacedPropertyValue = propertyValue;
+            for (String envName : envNames) {
+                String envValue = environment.get(envName);
+                if (envValue == null) {
+                    onError.accept(propertyKey, format("Configuration property '%s' references unset environment variable '%s'", propertyKey, envName));
+                    return;
+                }
+                replacedPropertyValue = replacedPropertyValue.replaceAll("\\$\\{ENV:" + envName + "}", envValue);
```

Comment:
Add a test case with

input
```
${ENV:FIRST}, ${ENV:SECOND}
```
where
FIRST is `first:${ENV:SECOND}:`
SECOND is `second:${ENV:FIRST}`

my expected results would be
```
first:${ENV:SECOND}:, second:${ENV:FIRST}
```

while current code would produce probably something like
```
first:second:${ENV:FIRST}:, second:${ENV:FIRST}
```

-- FIRST was first, so was apparently processed "recursively", while SECOND was not processed recursively